File: src/python/maps/map40.py
import ika

def AutoExec(engineRef):
    engineRef.background = ika.Image('gfx/mountains.png')

def to39(engineRef):
    offset_from = 8 * 16  # first vertical pos possible
    offset_to = 5 * 16  # first vertical pos possible
    x = engineRef.player.x - offset_from + offset_to
    yield from engineRef.mapSwitchTask('map39.ika-map', (x, 38 * 16))

File: src/winter/maps/map40.ts
import { MapScript } from "./mapscript.js"
import { PyEngine } from "./../winter.js"

export default new MapScript(autoexec, { to39 })

export function autoexec(engineRef: PyEngine) {
    const engine = engineRef.getEngine()
    engineRef.pySetBackground(engine.getImage('gfx/mountains.png'))
}

export function *to39(engineRef: PyEngine) {
    const offset_from = 8 * 16  // first vertical pos possible
    const offset_to = 5 * 16  // first vertical pos possible
    const p = engineRef.getPlayerEntity()
    const x = p.sprite.x - offset_from + offset_to
    yield* engineRef.mapSwitchTask('map39.ika-map', [x, 38 * 16])
}
